apps/explorer/lib/explorer/generic_paging_options.ex

```diff
@@ -27,26 +27,59 @@ defmodule BlockScoutWeb.GenericPagingOptions do
     }
   end
 
-  defp extract_page_size(%{"page_size" => page_size}, default_page_size) when is_bitstring(page_size) do
+  def extract_paging_options_from_params(
+        params,
+        max_page_size
+      ) do
+    %{
+      order_field: nil,
+      order_dir: nil,
+      page_size: extract_page_size(params, max_page_size),
+      page_number: extract_page_number(params)
+    }
+  end
+
+  defp extract_page_size(%{"page_size" => page_size}, max_page_size) when is_bitstring(page_size) do
     case Integer.parse(page_size) do
-      {page_size, _} -> page_size
-      :error -> default_page_size
+      {page_size, _} ->
+        if page_size > max_page_size or page_size < 1 do
+          max_page_size
+        else
+          page_size
+        end
+
+      :error ->
+        max_page_size
     end
   end
 
-  defp extract_page_size(_, default_page_size), do: default_page_size
+  defp extract_page_size(_, max_page_size), do: max_page_size
 
   defp extract_page_number(%{"page_number" => _}, 0), do: 1
 
   defp extract_page_number(%{"page_number" => page_number}, total_page_count) when is_bitstring(page_number) do
     case Integer.parse(page_number) do
-      {page_number, _} -> if page_number > total_page_count, do: min(total_page_count, page_number), else: page_number
-      :error -> 1
+      {page_number, _} ->
```

Comment:
we can add another case here to make it explicit and handle the condition where page_number > total_page_count

```elixir
case Integer.parse(page_number) do
      {page_number, _} when page_number > total_page_count ->
        min(total_page_count, page_number)

      {page_number, _}  ->
        page_number |> filter_non_negative_page_number()

       :error ->
         1 
end
```